### PR TITLE
initial content

### PR DESCRIPTION
**Pull request by andreaTP**
Hi there 👋 , sorry for the delay, but here you have the code of the `quarkus-kiota` extension extracted from https://github.com/kiota-community/kiota-java-extra/tree/main/quarkus (will remove the content there right after we have a first release from this repo).

@gastaldi I will need some help and support in setting up the GH Actions secrets etc. or do you have streamlined guidelines I should follow for the ci setup?

### CODE REVIEW

**Comment by gastaldi** [comment]
@andreaTP about the actions (in `.github` dir), I'd recommend copying from what's generated with the CLI when you run: `quarkus create extension kiota`

**Comment by andreaTP** [comment]
Done, I'm not sure if everything will work as expected at the first attempt but should be a decent start now.

File: .github/workflows/quarkus-snapshot.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  watch:
+    types: [started]
+
+  # For this CI to work, ECOSYSTEM_CI_TOKEN needs to contain a GitHub with rights to close the Quarkus issue that the user/bot has opened,
+  # while 'ECOSYSTEM_CI_REPO_PATH' needs to be set to the corresponding path in the 'quarkusio/quarkus-ecosystem-ci' repository
+
+env:
+  ECOSYSTEM_CI_REPO: quarkusio/quarkus-ecosystem-ci
+  ECOSYSTEM_CI_REPO_FILE: context.yaml
+  JAVA_VERSION: 11
```
Comment (gastaldi) [inline .github/workflows/quarkus-snapshot.yaml:13]:
```suggestion
  JAVA_VERSION: 17
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+name: Quarkiverse Release
```
Comment (gastaldi) [inline .github/workflows/release.yml:1]: This is outdated, can you make sure you have the latest Quarkus CLI installed?

**Review by gastaldi — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (andreaTP) [reply .github/workflows/release.yml:1]: you are right, it's outdated 😓 , let me iterate again

File: .github/release/maven-settings.xml.gpg
Comment (gastaldi) [inline .github/release/maven-settings.xml.gpg:1]: This file shouldn't be needed when the `.github/workflows/release.yml` workflow is up-to-date

**Comment by andreaTP** [comment]
Ready for a new round

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+# The '*' pattern is global owners.
+
+# Order is important. The last matching pattern has the most precedence.
+# The folders are ordered as follows:
+
+# In each subsection folders are ordered first by depth, then alphabetically.
+# This should make it easy to add new rules without breaking existing ones.
+
+*   @quarkiverse/quarkiverse-kiota-default
```
Comment (gastaldi) [inline .github/CODEOWNERS:14]:
```suggestion
*   @quarkiverse/quarkiverse-kiota
```

**Review by gastaldi — APPROVED** [review]
One minor thing, then it looks good

**Comment by andreaTP** [comment]
Thanks for the support @gastaldi do you wanna try to trigger a release?

**Comment by andreaTP** [comment]
and, do you mind to setup the "all-contributors" bot? or should I do it?

**Comment by gastaldi** [comment]
@all-contributors add @andreaTP for code and maintenance

**Comment by gastaldi** [comment]
> and, do you mind to setup the "all-contributors" bot? or should I do it?

Done, it's setup by default in the organization

**Comment by gastaldi** [comment]
> Thanks for the support @gastaldi do you wanna try to trigger a release?

Be my guest, here are the steps to release it: https://hub.quarkiverse.io/release/

**Comment by andreaTP** [comment]
I'll fix the CI and do a release asap

**Comment by andreaTP** [comment]
@gastaldi this seems to be on you:
https://github.com/quarkiverse/quarkus-kiota/actions/runs/7655896935/job/20863001078#step:8:2543

Would you mind having a look? 🙏

**Comment by gastaldi** [comment]
Your parent pom is missing the `<parent>` to quarkiverse-parent.

**Comment by andreaTP** [comment]
Thanks for checking 👍 fixing it

**Comment by andreaTP** [comment]
@gastaldi thanks for the support, finally a release went through so everything should be setup :-)

**Pull request merged**